File: src/main/java/top/litop/ptgame/config/ChannelConfig.java
package top.litop.ptgame.config;

import java.util.Map;

public class ChannelConfig {
    private String channelId;
    private String channelName;
    private Map<String, RoomConfig> gameRoomList;

    public class RoomConfig {
        private String roomId;
        private String roomName;

        public RoomConfig(String roomId, String roomName) {
            this.roomId = roomId;
            this.roomName = roomName;
        }

        public String getRoomId() {
            return roomId;
        }

        public String getRoomName() {
            return roomName;
        }

        public void setRoomName(String roomName) {
            this.roomName = roomName;
        }
    }

    public RoomConfig roomConfigBuilder(String roomId, String roomName) {
        return new RoomConfig(roomId, roomName);
    }

    public String getChannelId() {
        return channelId;
    }

    public String getChannelName() {
        return channelName;
    }

    public void setChannelName(String channelName) {
        this.channelName = channelName;
    }

    public Map<String, RoomConfig> getGameRoomList() {
        return gameRoomList;
    }

    public void setGameRoomList(RoomConfig roomConfig) {
        this.gameRoomList.put(roomConfig.getRoomId(), roomConfig);
    }

    public ChannelConfig(String channelId, String channelName) {
        this.channelId = channelId;
        this.channelName = channelName;
    }

}

File: ptgameback/ptgameback/src/main/kotlin/top/litop/ptgameback/config/ChannelConfig.kt
package top.litop.ptgameback.config

import top.litop.ptgameback.room.GameChannel

class ChannelConfig {
  public val channelId: String
  public var channelName: String

  class RoomConfig {
    public val roomId: String
    public var roomName: String
    public var eventLine: GameChannel

    constructor(roomId: String, roomName: String, eventLine: GameChannel) {
      this.roomId = roomId
      this.roomName = roomName
      this.eventLine = eventLine
      var eb = eventLine.vertx.eventBus()
      eb.publish("room." + eventLine.getGameChannelConfig() + "Channel", roomId + "房间:" + roomName + "被频道" + eventLine.getGameChannelConfig() + "创建")
    }
  }

  private var gameRoomList: MutableMap<String, RoomConfig>

  public fun getGameRoomList(): MutableMap<String, RoomConfig> {
    return this.gameRoomList
  }

  public fun setGameRoomList(roomId: String, roomConfig: RoomConfig) {
    this.gameRoomList.put(roomId, roomConfig)
  }


  constructor(channelId: String, channelName: String, eventLine: GameChannel) {
    this.channelId = channelId
    this.channelName = channelName
    this.gameRoomList = mutableMapOf<String, RoomConfig>()
  }
}
